### PR TITLE
Switch Linux to the GTK embedding

### PR DESCRIPTION
**Pull request by stuartmorgan-g**
## Description

Updates the tooling to use the GTK embedding, rather than the GLFW embedding:
- Adds new requirements to `doctor`
- Updates the app and plugin templates to make GTK-based runners and plugins
- Stops downloading and installing the GLFW artifacts

## Related Issues

Final part of https://github.com/flutter/flutter/issues/54860, other than cleanup.

## Tests

I added the following tests: New doctor functionality checks.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: https://flutter.dev/go/breaking-changes-template *Replace this with a link to a pull request that adds the migration guide to https://flutter.dev/docs/release/breaking-changes*
   - [x] Linux is not yet bound by the breaking change policy. The template version update will cause users to get instructions about updating their projects.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/linux/linux_doctor.dart [hunk 1]
```diff
@@ -43,13 +43,21 @@ class LinuxDoctorValidator extends DoctorValidator {
...
 
   final Map<String, Version> _requiredBinaryVersions = <String, Version>{
     kClangBinary: Version(3, 4, 0),
     kCmakeBinary: Version(3, 10, 0),
     kNinjaBinary: Version(1, 8, 0),
+    kPkgConfigBinary: Version(0, 29, 0),
   };
 
+  final List<String> _requiredLibraries = <String>[
+    'gtk+-3.0',
```
Comment (robert-ancell) [inline packages/flutter_tools/lib/src/linux/linux_doctor.dart:56]: Do we want to check the version of GTK here? If so this should be changed to `'gtk+-3.0 >= 3.22'` or similar. `pkg-config --exists gtk+-3.0 >= 3.22` or `pkg-config --exists gtk+3.0 --atleast-version=3.22` will return 1 if the version is not recent enough. You could use `--print-errors` to get error text or `--modversion if you want to do this yourself, i.e.

```
$ pkg-config --exists 'gtk+-3.0 >= 10' --print-errors
Requested 'gtk+-3.0 >= 10' but version of GTK+ is 3.24.18
```
```
$ pkg-config --modversion gtk+-3.0
3.24.18
```

I noticed we're not checking the version in the engine BUILD.gn - it should probably be done there too.

File: packages/flutter_tools/templates/app/linux.tmpl/CMakeLists.txt.tmpl [hunk 2]
```diff
@@ -28,14 +28,20 @@ set(FLUTTER_MANAGED_DIR "${CMAKE_CURRENT_SOURCE_DIR}/flutter")
 # Flutter library and tool build rules.
 add_subdirectory(${FLUTTER_MANAGED_DIR})
 
+# System-level dependencies.
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(GTK REQUIRED IMPORTED_TARGET gtk+-3.0)
```
Comment (robert-ancell) [inline packages/flutter_tools/templates/app/linux.tmpl/CMakeLists.txt.tmpl:33]: Again version information should be here.

File: packages/flutter_tools/templates/app/linux.tmpl/flutter/CMakeLists.txt [hunk 3]
```diff
@@ -20,7 +19,13 @@ function(list_prepend LIST_NAME PREFIX)
 endfunction()
 
 # === Flutter Library ===
-set(FLUTTER_LIBRARY "${EPHEMERAL_DIR}/libflutter_linux_glfw.so")
+# System-level dependencies.
+find_package(PkgConfig REQUIRED)
+pkg_check_modules(GTK REQUIRED IMPORTED_TARGET gtk+-3.0)
+pkg_check_modules(GLIB REQUIRED IMPORTED_TARGET glib-2.0)
+pkg_check_modules(GIO REQUIRED IMPORTED_TARGET gio-2.0)
```
Comment (robert-ancell) [inline packages/flutter_tools/templates/app/linux.tmpl/flutter/CMakeLists.txt:26]: Version information...

**Review by robert-ancell — APPROVED** [review]
LGTM. The versioning could be done in a follow up PR, as it will work OK currently, and just fail to build on older systems.

**Review by jonahwilliams — APPROVED** [review]
Tool changes LGTM overall,

Will leave template updates to @robert-ancell to review

**Comment by stuartmorgan** [comment]
> LGTM. The versioning could be done in a follow up PR, as it will work OK currently, and just fail to build on older systems.

Good idea, I didn't even think of library versions for some reason. I filed https://github.com/flutter/flutter/issues/59300 for that as a follow-up.

**Pull request merged**